src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotationMap.java

```diff
@@ -0,0 +1,208 @@
+            return null;
+        } else {
+            return values.get(0);
+        }
+    }
+
+    /**
+     * @return An empty FuncotationMap.  Never {@code null}
+     */
+    private static FuncotationMap createEmpty() {
+        return new FuncotationMap();
+    }
+
+    /**
+     * @return A FuncotationMap with only one transcript ID, which is {@see NO_TRANSCRIPT_AVAILABLE_KEY}.  Never {@code null}
+     */
+    public static FuncotationMap createNoTranscriptInfo(final List<Funcotation> funcotations) {
+        final FuncotationMap result = createEmpty();
+        result.add(NO_TRANSCRIPT_AVAILABLE_KEY, funcotations);
+        return result;
+    }
+
+    /**  This method checks for transcript duplications in the list of GencodeFuncotations.
+     * Recommended to gather all gencode funcotations before calling this method.
+     * @param gencodeFuncotations  Never {@code null}
+     * @return A new FuncotationMap created only from the given list of GencodeFuncotations.  Never {@code null}
+     */
+    public static FuncotationMap createFromGencodeFuncotations(final List<GencodeFuncotation> gencodeFuncotations) {
+
+        Utils.nonNull(gencodeFuncotations);
+        Utils.validateArg(!areDuplicateTranscriptIDsFound(gencodeFuncotations), "Duplicate transcript ID entries were found in input: " +
+                gencodeFuncotations.stream().map(gf -> gf.getAnnotationTranscript()).collect(Collectors.joining(",")));
+        final FuncotationMap result = createEmpty();
+        gencodeFuncotations.forEach(f -> result.add(f.getAnnotationTranscript(), f));
+        return result;
+    }
+
+    /**
+     * Add the given funcotations to the given transcript ID.
+     *
+     * @param txId  Never {@code null}
+     * @param funcotations  Never {@code null}
+     */
+    public void add(final String txId, final List<Funcotation> funcotations) {
+        Utils.nonNull(txId);
+        Utils.nonNull(funcotations);
+
+        final LinkedHashSet<Funcotation> existingFuncotationsToUpdate = txToFuncotations.getOrDefault(txId, new LinkedHashSet<>());
+        existingFuncotationsToUpdate.addAll(funcotations);
+        txToFuncotations.put(txId, existingFuncotationsToUpdate);
+    }
+    /** Add the given funcotation to the given transcript ID.
+     *
+     * @param txId Never {@code null}
+     * @param funcotation Never {@code null}
+     */
+    public void add(final String txId, final Funcotation funcotation) {
+        Utils.nonNull(txId);
+        Utils.nonNull(funcotation);
+        final LinkedHashSet<Funcotation> existingFuncotationsToUpdate = txToFuncotations.getOrDefault(txId, new LinkedHashSet<>());
+        existingFuncotationsToUpdate.add(funcotation);
+        txToFuncotations.put(txId, existingFuncotationsToUpdate);
+    }
+
+    /**
+     * Get the key list (i.e. transcript IDs) in order.
+     *
+     * @return Never {@code null}
+     */
+    public List<String> keyList() {
+        return new ArrayList<>(txToFuncotations.keySet());
```

Comment:
I renamed this to transcriptList, since that is what this is doing.  I wanted a type that would make it explicit that it was ordered.

If you would rather me make this a set (LinkedHashSet), I can do that.  I checked the other code and it tends to just iterate over the transcript IDs.  @jonn-smith 

Renamed... Done.

Reply:
@LeeTL1220 Even though we end up iterating most of the time, I think the flexibility of having the `LinkedHashSet` is worth adding.

Reply:
Done.